### PR TITLE
timeout as duration

### PR DESCRIPTION
**Pull request by baldersheim**
Conflicts:
	messagebus/src/vespa/messagebus/testlib/testserver.cpp

@vekterli PR
Look carefully at CommandQueue. Either I have misunderstood the unit of the timeout or there was a bug giving a too large timeout with a factor of 1000.

### CODE REVIEW

File: storage/src/vespa/storage/visiting/commandqueue.h [hunk 1]
```diff
@@ -141,11 +142,10 @@ CommandQueue<Command>::peekNextCommand() const
 
 template<class Command>
 void
-CommandQueue<Command>::add(
-        const std::shared_ptr<Command>& cmd)
+CommandQueue<Command>::add(const std::shared_ptr<Command>& cmd)
 {
     framework::MicroSecTime time(_clock.getTimeInMicros()
-            + framework::MicroSecTime(cmd->getQueueTimeout() * 1000000));
```
Comment (baldersheim) [inline storage/src/vespa/storage/visiting/commandqueue.h]: @vekterli This is where I suspect it adds it to the Q with a timeout that is 1000 X larger than was intended.

**Review by vekterli — APPROVED** [review]
:+1: great!

File: messagebus/src/vespa/messagebus/routablequeue.cpp [hunk 2]
```diff
@@ -39,17 +39,17 @@ RoutableQueue::enqueue(Routable::UP r)
 }
 
 Routable::UP
-RoutableQueue::dequeue(uint32_t msTimeout)
+RoutableQueue::dequeue(duration msTimeout)
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/routablequeue.cpp]: Fixed

File: messagebus/src/vespa/messagebus/routablequeue.cpp [hunk 3]
```diff
@@ -39,17 +39,17 @@ RoutableQueue::enqueue(Routable::UP r)
...
 {
     steady_clock::time_point startTime = steady_clock::now();
-    uint64_t msLeft = msTimeout;
+    duration msLeft = msTimeout;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/routablequeue.cpp]: Fixed

File: messagebus/src/vespa/messagebus/routablequeue.h [hunk 4]
```diff
@@ -69,7 +69,8 @@ class RoutableQueue : public IMessageHandler,
      * @return the dequeued routable
      * @param msTimeout how long to wait if the queue is empty
      **/
-    Routable::UP dequeue(uint32_t msTimeout);
+    Routable::UP dequeue(duration timeout);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/routablequeue.h]: Fixed

File: storage/src/vespa/storage/storageserver/documentapiconverter.cpp [hunk 5]
```diff
@@ -140,9 +138,12 @@ DocumentApiConverter::toStorageAPI(documentapi::DocumentMessage& fromMsg)
         break;
     }
 
-    if (toMsg.get() != 0) {
-        milliseconds timeout = std::min(milliseconds(INT_MAX), fromMsg.getTimeRemaining());
-        toMsg->setTimeout(timeout.count());
+    if (toMsg) {
+        //TODO getTimeRemainingNow ?
+        vespalib::duration cappedTimeout = (fromMsg.getTimeRemaining() < 1ms*INT_MAX)
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/storageserver/documentapiconverter.cpp]: Agree

**Pull request merged**